ci/provisioning/post_provision_config_nodes.sh

```diff
@@ -2,6 +2,53 @@
 
 set -eux
 
+# functions common to more than one distro specific provisioning
+url_to_repo() {
+    local url="$1"
+
+    local repo=${url#*://}
+    repo="${repo//%/}"
+    repo="${repo//\//_}"
+
+    echo "$repo"
+}
+
+add_repo() {
+    local repo="$1"
+    local gpg_check="${2:-true}"
+
+    if [ -n "$repo" ]; then
+        repo="${REPOSITORY_URL}${repo}"
+        if ! dnf repolist | grep "$(url_to_repo "$repo")"; then
+            dnf config-manager --add-repo="${repo}"
+            if ! $gpg_check; then
+                disable_gpg_check "$repo"
+            fi
+        fi
+    fi
+}
+
+disable_gpg_check() {
+    local url="$1"
+
+    repo="$(url_to_repo "$url")"
+    # bug in EL7 DNF: this needs to be enabled before it can be disabled
+    dnf config-manager --save --setopt="$repo".gpgcheck=1
+    dnf config-manager --save --setopt="$repo".gpgcheck=0
+    # but even that seems to be not enough, so just brute-force it
+    if [ -d /etc/yum.repos.d ] &&
+       ! grep gpgcheck /etc/yum.repos.d/"$repo".repo; then
+        echo "gpgcheck=0" >> /etc/yum.repos.d/"$repo".repo
+    fi
+}
+
+dump_repos() {
+        for file in $REPOS_DIR/*.repo; do
```

Comment:
(lint) Quote expansions in this for loop glob to prevent wordsplitting, e.g. "$dir"/*.txt . [SC2231]